python/feather/api.py

```diff
@@ -53,9 +53,11 @@ def read_dataframe(path, columns=None):
     data = {}
     names = []
     for i in range(reader.num_columns):
-        name, arr = reader.read_array(i)
-        data[name] = arr
-        names.append(name)
+        name = reader.get_column_name(i)
+        if (columns is None) or (name in columns):
```

Comment:
These parentheses are not necessary. You will also want to ensure that `columns` is converted to a set (otherwise you risk performing a bunch of unnecessary comparisons / linear scans of the columns list/array)